monkestation/code/modules/antagonists/mimic/mimic.dm

```diff
@@ -0,0 +1,398 @@
+
+	if(environment_smash)
+		EscapeConfinement()
+
+	if(AICanContinue(possible_targets))
+		var/atom/target_from = GET_TARGETS_FROM(src)
+		if(!QDELETED(target) && !target_from.Adjacent(target))
+			DestroyPathToTarget()
+		if(!MoveToTarget(possible_targets))     //if we lose our target
+			if(AIShouldSleep(possible_targets))	// we try to acquire a new one
+				toggle_ai(AI_IDLE)			// otherwise we go idle
+	return TRUE
+
+
+/mob/living/simple_animal/hostile/alien_mimic/attacked_by(obj/item/item, mob/living/target)
+	if(src in target.buckled_mobs) //Can't attack if its Got ya
+		return FALSE
+	..()
+
+/mob/living/simple_animal/hostile/alien_mimic/attack_hand(mob/living/target)
+	if(src in target.buckled_mobs)
+		return FALSE
+	if(disguised)
+		to_chat(target, "<span class='userdanger'>[src] latches onto you!</span>")
+		visible_message("<span class='danger'>[src] latches onto [target]!</span>",\
+				"<span class='userdanger'>You latch onto [target]!</span>", null, COMBAT_MESSAGE_RANGE)
+		latch(target)
+		restore()
+		toggle_ai(AI_ON)
+	else
+		..()
+
+/mob/living/simple_animal/hostile/alien_mimic/CanAttack(atom/the_target)
+	if(the_target == buckled)
+		return TRUE //fixes it jumping off of people immediately
+	if(iscarbon(the_target))
+		var/mob/living/carbon/carbon_target = the_target
+		if(carbon_target.stat == DEAD & should_heal() & !HAS_TRAIT(carbon_target, TRAIT_HUSK))
+			return TRUE
+	if(isliving(the_target))
+		var/mob/living/living_target = the_target
+		var/mob/living/simple_animal/hostile/alien_mimic/attacking_friend = locate() in living_target.buckled_mobs
+		if(attacking_friend & attacking_friend != src)
+			return FALSE
+		var/faction_check = faction_check_mob(living_target)
+		if((faction_check && !attack_same) || living_target.stat)
+			return FALSE
+	return TRUE
+
+/mob/living/simple_animal/hostile/alien_mimic/adjustHealth(amount, updating_health, forced)
+	if(amount > 0) //if you take damage, run
+		if(buckled)
+			resist_buckle()
+		if(!target)
+			FindTarget()
+		fleeing = TRUE
+	..()
+
+/mob/living/simple_animal/hostile/alien_mimic/AttackingTarget()
+	if(target == src) //Remove your disguise
+		restore()
+		return
+	if(isliving(target) & !buckled) //Latch onto people
+		var/mob/living/victim = target
+		if(iscarbon(victim) & victim.stat == DEAD & !HAS_TRAIT(victim, TRAIT_HUSK)) //Absorb someone to heal
+			visible_message("<span class='warning'>[src] starts absorbing [victim]!</span>", \
+						"<span class='userdanger'>You start absorbing [victim].</span>")
+			if(do_mob(src, victim, 10 SECONDS))
+				victim.become_husk(MIMIC_ABSORB)
+				people_absorbed++
+				adjustHealth(-30)
+		if(disguised) //Insta latch if youre disguised
+			latch(victim)
+			restore()
+			return
+		else if(do_mob(src, target, 3 SECONDS)) //Latch after a bit if you arent
+			latch(victim)
+			return
+	return ..() //If you're buckled, just attack normally
```

Comment:
This fires even if you're not buckled.
Rapidly clicking and moving will cause normal attacks